### PR TITLE
Fix Hebrew translation and BiDi support issues

### LINKED ISSUE #63: RTL: overlap in bottom panel/skin selector
## Bug description

As reported by @shanirub in https://github.com/rugk/awesome-emoji-picker/pull/62#issue-329930736:
> The small circle that represents the skin tone overlaps the text "Awesome Emoji" [in a RTL language].

## Steps to reproduce

1. (Use a version with the Hebrew or another RTL language translation.)
2.  Choose that language in the settings of Firefox.
3. Restart Firefox.
4. Open emoji picker.

### Actual behavior

awesome-emoji-picker-display-bug

emoji

### Expected behavior

* emoji should be at the right
* skin selector should be on the left

## System

Operating system and version: Fedora 30
Browser and version: Firefox 69
Add-on version: dev (master)

## Possible solution

likely CSS

Related issue: https://github.com/rugk/awesome-emoji-picker/issues/64

### LINKED ISSUE #64: RTL: truncated string in settings page when RTL/LTR is mixed
## Bug description

As reported by @shanirub in https://github.com/rugk/awesome-emoji-picker/pull/62#issue-329930736:
> Under the behaviour settings the "colon code / unicode" option is incorrectly displayed.

## Steps to reproduce

1. (Use a version with the Hebrew or another RTL language translation.)
2.  Choose that language in the settings of Firefox.
3. Restart Firefox.
4. Open settings of add-on.

### Actual behavior

awesome-emoji-picker-colon-bug

The `:` after (in the LTR sense) or before (in the RTL sense) the `colon` is missing.

### Expected behavior

* the colon should say `:colon:`

For comparison the German translation:
grafik

## System

Operating system and version: Fedora 30
Browser and version: Firefox 69
Add-on version: dev (master)

## Possible solution

When looking into the source HTML code, I can see, it (the text) is correctly applied/set. This likely is related to some display error related to RTL languages – or rather the mix of these.

Note the `:colon:` thing is in it's own `span` tag, because [this is how our translation system works](https://github.com/TinyWebEx/Localizer#keeping-child-elements-in-html).

grafik

```html
<label data-i18n="__MSG_optionCopyEmojiColons__" data-opt-i18n-keep-children="" for="resultType">השתמש ב- <code data-i18n="__MSG_optionCopyEmojiColonsCode__">:colon:</code> Code במקום ב- Unicode.</label>
```

Related issue: https://github.com/rugk/awesome-emoji-picker/issues/63

### LINKED ISSUE #70: Hebrew (he) translation is missing a placeholder
## Bug description

When submitting the add-on on https://addons.mozilla.org I noticed a warning that was shown me for one translation file.

## Steps to reproduce

You can likely run "[web-ext](https://github.com/mozilla/web-ext/) lint" to reproduce these warnings locally.

### Actual behavior

The warning is:
> ### Placeholder for message is missing
> **Warnung:** A placeholder used in the message is not defined.
> ```
> _locales/he/messages.json
> ```

### Expected behavior

No warning.

## System

Browser and version: Firefox 72
Add-on version: v1.2 / c0fa9b3

## Possible solution

I have no idea which placeholder it is, but somewhere one is not specified. E.g. the `$TEST$` is used in the message, but no placeholder replacement is defined. :thinking: 

See https://github.com/TinyWebEx/common/blob/master/CONTRIBUTING.md#translations for how these placeholders are translated, especially [the official doc](https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/Internationalization#Providing_localized_strings_in__locales).

/cc @shanirub (https://github.com/rugk/awesome-emoji-picker/pull/62)

### PR DESCRIPTION
**Pull request by danielrozenberg**
Fixes #63
Fixes #64
Fixes #70

### CODE REVIEW

**Comment by rugk** [comment]
Hi @danielrozenberg,
thanks for your first contribution to this project! :tada: :+1: 🏅
I hope you'll like this project and enjoy hacking on it… :smiley:

Thanks a lot also for targeting multiple related issues at once. :blush: 

I'll have a look at this soon.

> Note that the fix for #63 uses CSS attributes that are currently only supported in Firefox (inset-inline-start and inset-inline-end, which are BiDi-supportive alternatives to using left and right respectively), which is why I didn't send a separate pull request to emoji-mart

Yeah, okay, that seems to be unfortunate then, but it's good for now as this add-on is only available in Firefox _yet_ (see #54).
Remember you [could also use predefined strings such as `@@bidi_dir`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Internationalization#Predefined_messages), as the examples show also for padding or so. Though I'm not sure whether that works in other browser's (= Chrome/iums) WebExtension implementation…

---

Requesting a potential review/test from @shanirub (if you want). Unless there are objections, this will be merged soon.

To test out the code, you need to clone the repo locally (with git submodules) and load it into Firefox. Fortunately, that is quite easy without having to install additional tools. [In the contributing guide](https://github.com/TinyWebEx/common/blob/master/CONTRIBUTING.md#coding), you can see how to do that.

---

BTW if you care to be notified about translation updates subscribe to #27. :smiley:

Also notice there are more things to translate like the add-on listings on AMO (addons.mozilla.org) etc. [See the contributing guide for more information.](https://github.com/TinyWebEx/common/blob/master/CONTRIBUTING.md#translation-of-add-on-listings)

But as your PR is mostly a technical fix, I'll treat it as such going forward… :smiley:

**Comment by danielrozenberg** [comment]
> Remember you [could also use predefined strings such as `@@bidi_dir`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Internationalization#Predefined_messages), as the examples show also for padding or so. Though I'm not sure whether that works in other browser's (= Chrome/iums) WebExtension implementation…

Looks like `@@bidi_start_edge` is [supported in Chrome](https://developer.chrome.com/extensions/i18n#overview-predefined)  as well, unfortunately I won't be able to make this change and test it out before the weekend. Of course, you can merge my PR since the extension is Firefox only for now, and have it changed when it becomes relevant (or not changed if Chrome decides to support these attributes 😝)
 
> thanks for your first contribution to this project! 🎉 👍 🏅

my pleasure 😊

**Comment by rugk** [comment]
All right, we are not in a hurry (no release is planned or so in the next time), so I'll leave this PR open for now…

**Comment by danielrozenberg** [comment]
Updated to use the `@@bidi` messages:

LTR | RTL
--- | ---
Screen Shot 2020-08-01 at 15 06 53 | צילום מסך 2020-08-01 בשעה 14 59 43
Screen Shot 2020-08-01 at 15 07 07 | צילום מסך 2020-08-01 בשעה 15 03 20
Screen Shot 2020-08-01 at 15 07 17 | צילום מסך 2020-08-01 בשעה 15 03 29

File: src/common/common.css [hunk 1]
```diff
@@ -255,3 +255,38 @@ a:active {
   box-shadow: 0 0 0 1px var(--blue-50) inset, 0 0 0 1px var(--blue-50), 0 0 0 4px var(--blue-50-a30);
   border-radius: 2px;
 }
+
+/* Emoji Mart BiDi support */
```
[the remark this answers is not in the document]
Comment (danielrozenberg) [reply src/common/common.css]: Done!

**Comment by danielrozenberg** [comment]
Merge this whenever you feel like it 🙂

**Comment by rugk** [comment]
Okay, tested and looks fine to me…

grafik
grafik

(Except of the screenshot for the emoji search, but that currently is not localized anyway.)

**Pull request merged**